apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+    * @param _to Address to be approved
+    * @param _amount Number of `_arbitrationFeeToken` tokens to be approved
+    */
+    function _approveArbitratorFeeTokens(ERC20 _token, address _to, uint256 _amount) internal {
+        require(_token.safeApprove(_to, _amount), ERROR_TOKEN_APPROVAL_FAILED);
+    }
+
+    /**
+    * @dev Change Agreement settings
+    * @param _title String indicating a short description
+    * @param _content Link to a human-readable text that describes the initial rules for the Agreements instance
+    * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
+    */
+    function _newSetting(string _title, bytes _content, IArbitrator _arbitrator) internal {
+        require(isContract(address(_arbitrator)), ERROR_ARBITRATOR_NOT_CONTRACT);
+
+        uint256 id = settingsLength++;
+        settings[id] = Setting({ title: _title, content: _content, arbitrator: _arbitrator });
+        emit SettingChanged(id);
+    }
+
+    /**
+    * @dev Change the collateral requirements of a registered disputable app
+    * @param _disputable Disputable app
+    * @param _disputableInfo Disputable info instance for the disputable app
+    * @param _collateralToken Address of the ERC20 token to be used for collateral
+    * @param _actionAmount Amount of collateral tokens that will be locked every time an action is submitted
+    * @param _challengeAmount Amount of collateral tokens that will be locked every time an action is challenged
+    * @param _challengeDuration Challenge duration in seconds, during which the submitter can raise a dispute
+    */
+    function _changeCollateralRequirement(
+        IDisputable _disputable,
+        DisputableInfo storage _disputableInfo,
+        ERC20 _collateralToken,
+        uint256 _actionAmount,
+        uint256 _challengeAmount,
+        uint64 _challengeDuration
+    )
+        internal
+    {
+        require(isContract(address(_collateralToken)), ERROR_TOKEN_NOT_CONTRACT);
+
+        Staking staking = stakingFactory.getOrCreateInstance(_collateralToken);
+        uint256 id = _disputableInfo.collateralRequirementsLength++;
+        _disputableInfo.collateralRequirements[id] = CollateralRequirement({
+            token: _collateralToken,
+            staking: staking,
+            actionAmount: _actionAmount,
+            challengeAmount: _challengeAmount,
+            challengeDuration: _challengeDuration
+        });
+
+        emit CollateralRequirementChanged(_disputable, id);
+    }
+
+    /**
+    * @dev Tell whether an address has permission to challenge actions on a specific disputable app
+    * @param _disputable Disputable app being queried
+    * @param _challenger Address of the challenger
+    * @return True if the challenger can be challenge actions on the disputable app, false otherwise
+    */
+    function _canPerformChallenge(IDisputable _disputable, address _challenger) internal view returns (bool) {
+        if (!hasInitialized()) {
```

Comment:
Note that we theoretically don't need this check—everywhere we use it, it should already be checked by the action related to the disputable existing.